### ISSUE TITLE
Failed assertion in nghttp3_conn

### ISSUE THREAD

**Issue by jasnell** [ngtcp2/nghttp3#23]
Working on push stream support. When srclen is 0 and fin is 1, the following assertion fails.

```
Assertion failed: srclen, file c:\users\jasne\projects\quic\deps\nghttp3\lib\nghttp3_conn.c, line 573
```

It seems that possibly nghttp3_conn_read_uni is not handling `fin` correctly or am I missing something?

**Comment by tatsuhiro-t** [comment]
It looks like the current code does not handle this situation.  Will fix it.

**Comment by tatsuhiro-t** [comment]
Fixed this bug via 7b19d70d565f2995a57bc1ad0368f672f3fcea7b

**Comment by jasnell** [comment]
Thank you so much @tatsuhiro-t :-)
